core/src/main/java/com/netease/arctic/table/TableMetaStore.java

```diff
@@ -280,6 +280,7 @@ public synchronized UserGroupInformation getUGI() {
                   LOG.warn("Fail to reflect UserGroupInformation", e);
                 }
               }
+              System.setProperty("sun.security.krb5.principal", krbPrincipal);
```

Comment:
To avoid side effects, I think we should 
- not set system property if it is not an unexpected value
- set the system property back to the old value after relogin

The process may be like:
```
String oldSystemPrincipal = System.getProperty("sun.security.krb5.principal");
boolean systemPrincipalChanged = false;
try {
    ...

    if (oldSystemPrincipal != null && !krbPrincipal.equals(oldSystemPrincipal)) {
        System.setProperty("sun.security.krb5.principal", krbPrincipal);
        systemPrincipalChanged = true;
    }
    ugi.checkTGTAndReloginFromKeytab();
} finally {
    ...

    if (systemPrincipalChanged) {
        System.setProperty("sun.security.krb5.principal", oldSystemPrincipal);
    }
}
```

Reply:
I agree with the first, but I have a little concern about the second.

> set the system property back to the old value after relogin

`oldSystemPrincipal == null` cause an exception to occur, although I don't know exactly what caused it.
After relogin reset to `null`, the next relogin will get the same value `null`?

Reply:
`multiple catalogs with kerberos` is one of the situations we have to handle. 

Besides,  the old value of `sun.security.krb5.principal` is very likely to be used by another authentication process in this JVM, even though we don't know what they are, they may fail if we change the old value.

Reply:
> `multiple catalogs with kerberos` is one of the situations we have to handle.
> 
> Besides, the old value of `sun.security.krb5.principal` is very likely to be used by another authentication process in this JVM, even though we don't know what they are, they may fail if we change the old value.

I get

Reply:
BTW, I delete `oldSystemPrincipal != null` because oldSystemPrincipal is null in my test case.

Reply:
Thanks for your quick reply!

I have reviewed the changes, but I still think we should keep the `oldSystemPrincipal != null`. The reason is in most cases, the `sun.security.krb5.principal` is supposed to be null, and it's not necessary to set our `krbPrincipal` into it for these cases.

Also, to avoid the risk of `NullPointerException`, we'd better code it like this

```
    if (oldSystemPrincipal != null && oldSystemPrincipal.equals(krbPrincipal)) {
        System.setProperty("sun.security.krb5.principal", krbPrincipal);
        systemPrincipalChanged = true;
    }
```